services/infinispan-device-registry/src/main/java/org/eclipse/hono/deviceregistry/RegistryCredentialObject.java

```diff
@@ -0,0 +1,50 @@
+/*******************************************************************************
+ * Copyright (c) 2016, 2018 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package org.eclipse.hono.deviceregistry;
+
+import io.vertx.core.json.JsonObject;
+import org.eclipse.hono.util.CredentialsObject;
+
+/**
+ * A custom class to be used as value in the backend key-value storage.
+ * This store credentials details.
+ *
+ *  See {@link org.eclipse.hono.deviceregistry.CacheTenantService CacheTenantService} class.
+ */
+public class RegistryCredentialObject {
+
+    private final String tenantId;
```

Comment:
what is this good for if it is not used anywhere?

Reply:
When doing a batch request (i.e. `getAll` or `removeAll`) the value is used as a field to do a query against the cache.
See `queryAllCredentialsForDevice(String, String)` in `CredentialCache`